components/inputs/input-textarea.js

```diff
@@ -0,0 +1,261 @@
+		const disabled = this.disabled || this.skeleton;
+
+		const mirrorStyles = {};
+		if (this.minHeight) mirrorStyles.minHeight = this.minHeight;
+		if (this.maxHeight) mirrorStyles.maxHeight = this.maxHeight;
+
+		const textarea = html`
+			<div class="d2l-input-textarea-container d2l-skeletize">
+				<div class="d2l-input d2l-input-textarea-mirror" style="${styleMap(mirrorStyles)}" aria-invalid="${ifDefined(ariaInvalid)}">
+					${lines.map(line => html`${line}<br />`)}
+				</div>
+				<textarea aria-describedby="${ifDefined(this.description ? this._descriptionId : undefined)}"
+					aria-invalid="${ifDefined(ariaInvalid)}"
+					aria-label="${ifDefined(this._getAriaLabel())}"
+					aria-required="${ifDefined(ariaRequired)}"
+					@blur="${this._handleBlur}"
+					@change="${this._handleChange}"
+					class="d2l-input"
+					?disabled="${disabled}"
+					id="${this._textareaId}"
+					@input="${this._handleInput}"
+					@invalid="${this._handleInvalid}"
+					maxlength="${ifDefined(this.maxlength)}"
+					minlength="${ifDefined(this.minlength)}"
+					placeholder="${ifDefined(this.placeholder)}"
+					?required="${this.required}">${this.value}</textarea>
+				${this.validationError ? html`<d2l-tooltip for=${this._textareaId} state="error" align="start">${this.validationError}</d2l-tooltip>` : null}
+			</div>
+			${offscreenContainer}
+		`;
+
+		if (this.label && !this.labelHidden) {
+			return html`
+				<label class="d2l-input-label d2l-skeletize" for="${this._textareaId}">${this.label}</label>
+				${textarea}`;
+		}
+
+		return textarea;
+
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === 'value') {
+				this.setValidity({ tooShort: this.minlength && this.value.length > 0 && this.value.length < this.minlength });
+				this.requestValidate(false);
+				this.setFormValue(this.value);
+				this._prevValue = (oldVal === undefined) ? '' : oldVal;
+			} else if (prop === 'validationError') {
+				if (oldVal && this.validationError) {
+					const tooltip = this.shadowRoot.querySelector('d2l-tooltip');
+					tooltip.updatePosition();
+				}
+			}
+		});
+	}
+
+	async focus() {
+		const elem = this.shadowRoot.querySelector('textarea');
+		if (elem) {
+			elem.focus();
+		} else {
+			await this.updateComplete;
+			this.focus();
+		}
+	}
+
+	get validationMessage() {
+		if (this.validity.tooShort) {
+			return this.localize('components.form-element.input.text.tooShort', { label: this.label, minlength: formatNumber(this.minlength) });
+		}
+		return super.validationMessage;
+	}
+
+	get validity() {
+		const elem = this.shadowRoot.querySelector('textarea');
+		if (!elem.validity.valid) {
+			return elem.validity;
+		}
+		return super.validity;
+	}
+
+	_getAriaLabel() {
+		if (this.label && this.labelHidden) {
+			return this.label;
+		}
+		if (this.hasAttribute('aria-label')) {
```

Comment:
Supporting `aria-label` was a backwards-compatibility thing that we had to do for `<d2l-input-text>` because it didn't initially support `label-hidden`. No need to support it here.